### PR TITLE
Express Route - part 1

### PR DESCRIPTION
**Pull request by lenala**

### CODE REVIEW

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+     */
+    @Method
+    ExpressRouteCircuitPeerings peerings();
+
+    // Getters
+
+    /**
+     * @return the SKU
+     */
+    ExpressRouteCircuitSku sku();
```
Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java]: I guess this is auto-gened class, which contains `tier` and `family`. Usually we create an enum for such skus.. For example an enum for this type can have the values

```java
ExpressRouteCircuitSkuTypes.STANDARD_UNLIMITEDDATA
ExpressRouteCircuitSkuTypes.STANDARD_METEREDDATA
ExpressRouteCircuitSkuTypes.PREMIUM_UNLIMITEDDATA
ExpressRouteCircuitSkuTypes.PREMIUM_METEREDDATA
```

where _ separates skuTier and family.

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        interface WithSkuTier {
+            WithSkuFamily withSkuTier(ExpressRouteCircuitSkuTier skuTier);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify SKU family. Possible values include: 'UnlimitedData', 'MeteredData'.
+         */
+        interface WithSkuFamily {
+            WithCreate withSkuFamily(ExpressRouteCircuitSkuFamily skuFamily);
+        }
```
Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java]: If we define an enum `ExpressRouteCircuitSkuTypes` as mentioned in the previous comment then we can have a single wither `withSkuType(ExpressRouteCircuitSkuTypes sku)` that replaces `withSkuTier` and `WithSkuFamily`

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+         */
+        interface WithServiceProvider {
+            WithPeeringLocation withServiceProvidet(String serviceProviderName);
+        }
+
+        /**
+         * The stage of express route circuit definition allowing to specify service provider peering location.
+         */
+        interface WithPeeringLocation {
+            WithBandwidth withPeeringLocation(String location);
```
[the remark this answers is not in the document]
Comment (lenala) [reply azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java]: No, this is not same. There is actually API that allows to get a list of service providers and available locations and bandwidth for each of them which is not exposed yet through fluent java api, auto-generated is in ExpressRouteServiceProviderInner. But location is completely different from region.

[the remark this answers is not in the document]
Comment (anuchandy) [reply azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/ExpressRouteCircuit.java]: Thanks for confirming

**Pull request merged**